### DISCUSSION TITLE
Subheading and nested list indenting

### DISCUSSION THREAD

**Discussion by jghauser** [nvim-neorg/neorg#411] [Q&A] [answered]
I loooooove the new indent feature, thanks a lot! :partying_face: I have one small question regarding it. Is it possible to have nested lists and subheadings indent one character more than the list/heading at the level above. So instead of this:
```
- an item
-- a subitem

* a heading
** a subheading
```
I would like this:
```
- an item
 -- a subitem

* a heading
 ** a subheading
```
The advantage of this is that (a) it looks neat as bullet points are under the first letter of the item above and (b) it makes (2 space) tab always line up with the beginning of lists/headings.

**Comment by vhyrro** [comment, marked as answer]
Hey! Glad you like it! It's an initial push, so there's not a whole lot of out of the box configuration, but I'll be adding a special configuration table tomorrow that'll allow you to configure how much extra you want to indent for different syntax elements, so you can configure it to your tastes. There's also indent presets on my TODO list, but I think that's more of a thing for 0.2 :P

**Reply by jghauser** [reply]
Thanks for the answer! Looking forward to the config table :)

**Reply by vhyrro** [reply]
As all things technology, bugs got in the way :P

There was an internal issue with the indent engine that simply returned the wrong nodes on certain lines which was worked around using hacks - this prevented easy customizability, because to e.g. indent code block contents differently you'd have to modify the indent level for `_line_break`, which didn't make much sense :sweat_smile:. That's been fixed now though! All that's left is implementing that special config table I initially promised and we'll be set.

**Reply by jghauser** [reply]
Haha that sounds NOT like fun. No worries, I totally understand! I'm already very grateful for the huge amount of work you must be investing into neorg! :)

**Reply by vhyrro** [reply]
Thank you! The table should be complete now. It's very simple, in your case it would be something like this:
```
["core.norg.esupports.indent"] = {
	config = {
		tweaks = {
			heading2 = 1,
			heading3 = 2,
			heading4 = 3,
			heading5 = 4,
			heading6 = 5,
		}
	}
}
```
Note that there are limitations right now that I'll be working on in the future:
- Tweaks only increment/decrement the amount that should be indented, but don't change the indent amount to a static value. This means that what the above code snippet does is: indent a heading1 as you would normally, then add one to the final number of spaces to indent.
  This is intended. To change the actual indent logic for a certain node type, change the `indents` table instead.
- Nodes that are set to "autoindent" (this is currently only done for content within ranged tags) cannot be tweaked.
- Headings that are above level 6 will not indent properly with tweaks, as everything after 6 asterisks falls back to a `heading6` node.
  Perhaps this can be worked around, but I'm unsure :p

Still, it should do what you want! If you find any other bugs then be sure to make an issue for those :)

**Reply by jghauser** [reply]
This is perfect! I've added some entries for ordered and unordered lists and now everything is indenting exactly as i want it to. Thanks! :)
